src/utils.ts

```diff
@@ -13,6 +13,18 @@ export const getEpochWithMillis = (time: number): number => {
   return time < 100000000000 ? time * 1000 : time;
 };
 
+/**
+ * This formats a time field to ensure the millisecond portion of the epoch time is
+ * NOT included.
+ *
+ * @param time the epoch time to format
+ *
+ * @returns the formatted epoch time containing milliseconds.
+ */
+export const getEpochWithoutMillis = (time: number): number => {
+  return time > 100000000000 ? time % 1000 : time;
```

Comment:
This seems a bit odd, what's it for?

Reply:
Yes, it is. I wanted the ability to support time stamps with and without milliseconds. _Grafana_ time ranges are represented without milliseconds, but the query parameter requires them. These are utility methods to convert the timestamps from the datasource query to/from to the required format. Ideally, we should strongly type the to/from early. Maybe the next release? I wanted to make sure this made it into the open _Grafana_ PR.

Reply:
Wait, are you saying that they both have a strictly defined timestamp, or that it's _sometimes_ milliseconds and _sometimes_ seconds? Or are you saying that's changing in a future Grafana version?

Reply:
The `data.timeRange.from` and `data.timeRange.to` received by the plugin are typed as `DateTime` (from `Moment`). There is 
a wrapper function to get the epoch. This function returns without millis. The query parameter used to update the time range in `getLocationSrv().update()` requires a date/time in Epoch (with milliseconds) format.

Reply:
I'm just not happy with the `time > 100000000000` check, it's a magic constant, really.

Reply:
I agree. This was just the easiest way to determine if the number is in milliseconds or seconds as an Epoch can be formatted with either only seconds or milliseconds. I looked to see if the `moment` library contained any utilities for this, but I did not see any. I am certainly open to a different method.

Reply:
I think we should only be accepting UNIX timestamps of seconds for `time_from` and `time_to` - there's really no way to detect whether the user has provided ms or s. It has to be a set unit. Probably should have brought this up in a previous PR - my apologies.

Reply:
I don't know. This may be OK for non timestamp fields, but if the user wants to use the timestamp field as a `time_from` or `time_to` some TSDB provide timestamps in seconds and some provide in ms. Mandating the datasource to provide only in seconds requires the user to transform the returned datasource before passing it to the plugin. This is a challenge as some (`InfluxDB`) do not allow direct manipulation of the timestamp field in a normal query.

The current check while not graceful is able to determine if the time is in seconds or ms based on the length of the field. This allows the plugin to operate on either.

Reply:
Alright, I guess someone 1000 years in the future can fix it. :)

Reply:
Haha. Thanks! We can always revisit.